controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +181,59 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) createOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	} else if secret.UID == "" {
+		if err := r.own(secret); err != nil {
+			return err
+		}
+		storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+		if err != nil {
+			return err
+		}
+		token, err := util.GenerateClientOnboardingToken(
+			r.TokenLifetimeInHours,
+			onboardingPrivateKeyFilePath,
+			storageCluster.UID,
+			r.storageConsumer.Name,
+		)
+		if err != nil {
+			return err
+		}
+		secret.Data = map[string][]byte{
+			"ticket": []byte(token),
+		}
+		util.AddAnnotation(secret, "token-expiry-utc", time.Now().Add(time.Duration(r.TokenLifetimeInHours)*time.Hour).UTC().String())
+		r.Log.Info("creating secret containing onboarding token for", "StorageConsumer", r.storageConsumer.Name, "Secret", secret.Name)
+		if err := r.Create(r.ctx, secret); err != nil {
```

Comment:
Where is the owner ref?

Reply:
the first step in this condition was adding the owner L191.

Reply:
deferred.